cdc/processor/pipeline/table_actor.go

```diff
@@ -0,0 +1,351 @@
+	t.nodes = append(t.nodes, NewActorNode(c, d))
+
+	t.started = true
+	log.Info("table actor is started", zap.String("tableName", t.tableName), zap.Int64("tableID", t.tableID))
+	return nil
+}
+
+func (t *tableActor) stop(err error) {
+	if t.stopped {
+		return
+	}
+	t.stopped = true
+	t.err = err
+	t.cancel()
+	log.Info("table actor will be stopped",
+		zap.String("tableName", t.tableName), zap.Int64("tableID", t.tableID), zap.Error(err))
+}
+
+func (t *tableActor) checkError() {
+	if t.err != nil {
+		t.reportErr(t.err)
+		t.err = nil
+	}
+}
+
+// ============ Implement TablePipline, must be threadsafe ============
+
+// ResolvedTs returns the resolved ts in this table pipeline
+func (t *tableActor) ResolvedTs() model.Ts {
+	// TODO: after TiCDC introduces p2p based resolved ts mechanism, TiCDC nodes
+	// will be able to cooperate replication status directly. Then we will add
+	// another replication barrier for consistent replication instead of reusing
+	// the global resolved-ts.
+	if redo.IsConsistentEnabled(t.replConfig.Consistent.Level) {
+		return t.sinkNode.ResolvedTs()
+	}
+	return t.sortNode.ResolvedTs()
+}
+
+// CheckpointTs returns the checkpoint ts in this table pipeline
+func (t *tableActor) CheckpointTs() model.Ts {
+	return t.sinkNode.CheckpointTs()
+}
+
+// UpdateBarrierTs updates the barrier ts in this table pipeline
+func (t *tableActor) UpdateBarrierTs(ts model.Ts) {
+	msg := message.BarrierMessage(ts)
+	err := t.tableActorRouter.Send(t.actorID, msg)
+	if err != nil {
+		log.Warn("send fails", zap.Reflect("msg", msg), zap.String("tableName", t.tableName), zap.Int64("tableID", t.tableID), zap.Error(err))
+	}
+}
+
+// AsyncStop tells the pipeline to stop, and returns true is the pipeline is already stopped.
+func (t *tableActor) AsyncStop(targetTs model.Ts) bool {
+	msg := message.StopSinkMessage()
+	err := t.tableActorRouter.Send(t.actorID, msg)
+	log.Info("send async stop signal to table", zap.String("tableName", t.tableName), zap.Int64("tableID", t.tableID), zap.Uint64("targetTs", targetTs))
+	if err != nil {
+		if cerror.ErrMailboxFull.Equal(err) {
+			return false
+		}
+		if cerror.ErrSendToClosedPipeline.Equal(err) {
+			return true
+		}
+		log.Panic("send fails", zap.Reflect("msg", msg), zap.Error(err))
+	}
+	return true
+}
+
+// Workload returns the workload of this table
+func (t *tableActor) Workload() model.WorkloadInfo {
+	// We temporarily set the value to constant 1
+	return workload
+}
+
+// Status returns the status of this table pipeline
+func (t *tableActor) Status() TableStatus {
+	return t.sinkNode.Status()
+}
+
+// ID returns the ID of source table and mark table
+func (t *tableActor) ID() (tableID, markTableID int64) {
+	return t.tableID, t.markTableID
+}
+
+// Name returns the quoted schema and table name
+func (t *tableActor) Name() string {
+	return t.tableName
+}
+
+// Cancel stops this table actor immediately and destroy all resources
+// created by this table pipeline
+func (t *tableActor) Cancel() {
```

Comment:
Shall the caller always calls `Wait()` after it calls `Cancel()` to wait for the actor truly stop?

Reply:
When should call Cancel() or AsyncStop() ?

Reply:
caller call AsyncStop to release sink, and then call Cancel and wait to stop the pipeline.